src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,124 @@
+
+    /**
+     * Method that find all the information of user entity
+     * @return: a collection type list with the information of user table
+     */
+    @GetMapping
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * Method that find information by primary key
+     * @param id : primary key of one table
+     * @return: the information of with  id you send.
+     */
+    @GetMapping("/{id}")
+    public User findOne(@PathVariable Long id) throws UserNotFoundException {
+        return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(
+                ErrorConstants.NOT_EXIST_USER_ID));
+    }
+
+    /**
+     * Method that save the fields from an entity.
+     * @param user object of entity user
+     * @return: null(message of success or warning )
+     */
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    /**
+     * Method that delete a registry of one entity by primary key
+     * @param id :primary key of one table
+     */
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(
+                ErrorConstants.NOT_EXIST_USER_ID));
+        userRepository.deleteById(id);
+    }
+
+    /**
+     * method that update a registry by primary key
+     * @param user : object of entity user
+     * @param id : primary key of one table
+     * @return null(message of success or warning )
+     */
+    @PutMapping("/{id}")
+    public User updateBook(@RequestBody User user, @PathVariable Long id) throws UserIdMismatchException {
+        if (user.getId() != id) {
+            throw new UserIdMismatchException();
+        }
+        userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(
+                ErrorConstants.NOT_EXIST_USER_ID));
+        return userRepository.save(user);
+    }
+    /**
+     * method that save a registry a book
+     * @param id: user_id from user table
+     * @param book: object book to save
+     * @return null(message of success or warning )
+     * @throws UserNotFoundException
+     */
+    @PutMapping("/{id}/books")
+    @ResponseStatus(HttpStatus.CREATED)
+    public User addBook(@PathVariable Long id, @RequestBody Book book)
+            throws UserNotFoundException {
+        User user = userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(
+                ErrorConstants.NOT_EXIST_USER_ID));
+        user.addBook(book);
+        return userRepository.save(user);
+    }
+
+    /**
+     * method removes a registry from book table
+     * @param id: user_id from user table
+     * @param book: object book to remove
+     * @return
+     * @throws UserNotFoundException
+     */
+    @DeleteMapping("/{id}/books")
+    public User deleteBook(@PathVariable Long id, @RequestBody Book book)
```

Comment:
I would say that deleteBook from the users list would be a PUT instead of a DELETE, since you are not deleting anything but only editing the user's book list.

When making this change you will surely have problems with the url since they would be the same.
You could change the path only or change the path and change the variable path by a request param.

Reply:
I made the modification in this issue.